apps/user-ms/src/main/java/it/pagopa/selfcare/user/controller/UserController.java

```diff
@@ -319,6 +320,29 @@ public Uni<Response> createOrUpdateByUserId(@PathParam("userId") String userId,
 
     }
 
+    /**
+     *
+     * @param userId  String
+     * @param userDto AddUserRoleDto
+     */
+    @Operation(description = "Checks if the user is already a manager for the specified product and, if not, creates or updates the user with a new role.", summary = "Check if the user is manager or Update/create a user by userId with a new role")
+    @APIResponses({
+            @APIResponse(responseCode = "200", description = "The user is already a manager for the specified product."),
+            @APIResponse(responseCode = "201", description = "The user has been created or updated with a new role."),
+    })
+    @POST
+    @Path("/{userId}/onboarding")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Uni<Response> createUserByUserId(@PathParam("userId") String userId,
+                                            @Valid AddUserRoleDto userDto,
+                                            @Context SecurityContext ctx) {
+        return readUserIdFromToken(ctx)
+                .onItem().transformToUni(loggedUser -> userService.createUserByUserId(userDto, userId, loggedUser))
+                .onItem().transform(ignore -> Response.status(HttpStatus.SC_CREATED).build())
```

Comment:
the status code 200 or 201 expected that the response contains a body....you could add User object as body in both case